src/asmcnc/skavaUI/screen_job_recovery.py

```diff
@@ -476,18 +476,21 @@ def next_screen(self):
             self.wait_popup = popup_info.PopupWait(self.sm, self.l)
             self.jd.job_recovery_selected_line = self.selected_line_index + 1
             self.go_xy()
-            Clock.schedule_once(self.proceed_to_next_screen, 0.4)
+            Clock.schedule_once(self.wait_for_idle, 0.4)
         else:
             error_message = self.l.get_str('Please ensure machine is idle before continuing.')
             popup_info.PopupError(self.sm, self.l, error_message)
 
-    def proceed_to_next_screen(self, dt):
+    def wait_for_idle(self, dt):
```

Comment:
probably need some kind of break condition or event handling here; if there is an alarm or door or something, then that screen will come up, but the clocks here will continue in the background and once the machine state goes to idle then it will bring up the nudge screen, potentially at a surprising point :)